### PR TITLE
Handle the staged events queue being empty.

### PR DESCRIPTION
**Pull request by clokep**
(I'm not 100% sure the title is correct.)

This attempts to fix https://sentry.matrix.org/sentry/synapse-matrixorg/issues/225383/ by properly handling the return value from `get_next_staged_event_id_for_room`, but I'm not 100% sure it is doing the right thing. :)

### CODE REVIEW

File: synapse/federation/federation_server.py [hunk 1]
```diff
@@ -972,13 +972,21 @@ async def _process_incoming_pdus_in_room_inner(
...
-            if next_origin != latest_origin or next_event_id != latest_event.event_id:
+            result = await self.store.get_next_staged_event_id_for_room(room_id)
+            if result is None:
                 latest_origin = None
                 latest_event = None
+            else:
+                (
+                    next_origin,
+                    next_event_id,
+                ) = result
```
Comment (erikjohnston) [inline synapse/federation/federation_server.py:983]:
```suggestion
                next_origin, next_event_id  = result
```

**Review by erikjohnston — APPROVED** [review]
Yeah that looks about right! I wonder if we should just return immediately if we get a `None` back, rather than doing `get_next_staged_event_for_room` (which does the same)?

File: synapse/federation/federation_server.py (hunk 1 above)
Comment (clokep) [reply synapse/federation/federation_server.py:983]: I'm surprised lint was happy with this. I think I assumed black would just reformat it. Oops!

**Comment by clokep** [comment]
> Yeah that looks about right! I wonder if we should just return immediately if we get a `None` back, rather than doing `get_next_staged_event_for_room` (which does the same)?

I don't know, I just kind of followed the code that was there (I found this function quite hard to follow TBH).

I should probably have mentioned that `mypy` would have caught this if we were checking the corresponding storage file with mypy, but that has many errors when I added it.
